.travis.yml

```diff
@@ -15,8 +18,9 @@ install:
 script:
   # install the dependencies
   - run install
-  # generate and report the test coverage
-  - run tests:coverage
+  # generate and report the test coverage if we have a token for coveralls
+  - 'if [ "$COVERALLS_TOKEN" != "" ]; then run tests:coverage; fi'
+  - 'if [ "$COVERALLS_TOKEN" = "" ]; then run tests; fi'
```

Comment:
I think this needs to be `==`

Reply:
So I pulled this from [this doc page](https://docs.travis-ci.com/user/pull-requests/) on travis and it does look like the correct test script was run if I look at the travis build. That being said, I'm definitely far from a bash expert (didn't even occur to me to use `-z` or `-n`) so you could definitely be correct.

Reply:
I did some digging and they are both valid equality operators. `=` works with the `test [` command and `==` works inside of `[[` commands

Reply:
Ah you are right, I didn't know that -- definitely not a bash expert either.